exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillSqlWorker.java

```diff
@@ -131,7 +130,9 @@ private static PhysicalPlan convertPlan(QueryContext context, String sql, Pointe
           "Will sync remote and local function registries if needed and retry " +
           "in case if issue was due to missing function implementation.", e);
       if (context.getFunctionRegistry().syncWithRemoteRegistry(
-          context.getDrillOperatorTable().getFunctionRegistryVersion())) {
+              context.getDrillOperatorTable().getFunctionRegistryVersion())
```

Comment:
Please explain? What about CTAS for example?

Reply:
Thanks, `CTAS` with `EXPLAIN` also should be present in this check. The goal was to avoid rerunning queries that cannot have UDFs but failed with an exception. For example, for Metastore ANALYZE, Metastore is being disabled using a query-level option before plan constructing but after the check that it was enabled. So after rerunning such a query, it will fail with another exception that Metastore is disabled.